presets/content.nim

```diff
@@ -32,66 +32,175 @@ macro defineContentTypes*(body: untyped): untyped =
+          f.check nnkCall
+          f.check 2
+
+          let
+            fName = f[0]
+            fValue = f[1]
+          
+          fName.check nnkIdent
+          fValue.check nnkStmtList
+          fValue.check 1
+
+          constr.add(fName.colon(fValue[0]))
+
+      
+      oName.assign constr
+
+      case maybeInit.kind:
+      of nnkStmtList:
         discard
-    initBody = initProc[6]
+      of nnkCall:
+        maybeInit.insert(1, oName)
+        inits.add maybeInit
+      of nnkIdent:
+        inits.add newTree(nnkCall, maybeInit, oName)
+      else:
+        error "illegal initializer", oName
+    else:
+      error "illegal object declaration", o
+  
+  inits.add quote do: 
+    `listName`.add `listValues`
+  vars.add decl(listName, quote do: seq[`name`])
 
-  result.add letSec
-  result.add initProc
-  var usedNames: HashSet[string]
+## Syntax sugar for defining and initializing content of the game.
+## Way you define objects inside a macro reduces amount of text you 
+## have to write and makes content more readable. example:
+## 
+##  ..block::nim
+##  makeContent:
+##    CustomTypeName:
+##      objectName
+##      ## or
+##      objectName initializer
+##      ## or
+##      objectName initializer(argument)
+##      ## or 
+##      objectName:
+##        fieldName: value
+##      ## or case 2 and 3 combined with 4
+## 
+##  
+## TODO: The syntax does not propagate into nested expression.
+## All content is stored in its own list and also all content goes to 
+## `all` list. Every structure you include has to inherit `Content`
+macro makeContent*(body: untyped): untyped =
+  body.check nnkStmtList
+
+  var 
+    id: int
+    inits = newStmtList()
+    all = newNimNode(nnkBracket)
+    vars = newNimNode(nnkVarSection)
+
+  result = newStmtList()
 
+  for s in body:
+    s.check nnkCall
+    s.check 2
+
+    let
+      sName = s[0]
+      sBody = s[1]
+    
+    sName.check nnkIdent
+    sBody.check nnkStmtList
+
+    parseSection(sName, sBody, id, all, vars, inits)
+  
+  let a = ident("all") # macro is buggy
+  inits.add quote do: 
+    `a`.extend `all`
+  
+  result.add vars
+  result.add quote do: 
+    var all {.inject.}: seq[Content]
   result.add quote do:
-    var contentList* {.inject.}: seq[Content]
-
-  for n in body:
-    if n.kind == nnkAsgn:
-      var 
-        nameIdent = $n[0] #name of content
-        consn = n[1] #object constructor call
-        typet = consn[0]
-        typeName = $consn[0] #e.g. "Block"
-        listName = ident(typeName.toLowerAscii & "List")
-        resName = typeName.toLowerAscii & nameIdent.capitalizeAscii
-      
-      if typeName notin usedNames:
-        result.add quote do:
-          var `listName`* {.inject.}: seq[`typet`]
-        usedNames.incl typeName
-      
-      #switch empty calls to constructors
-      if consn.kind == nnkCall:
-        consn = newNimNode(nnkObjConstr).add(ident(typeName))
-      
-      defaultValues.withValue(typeName, defs):
```

Comment:
I don't see a replacement for `defaultValues` anywhere; why was it removed?

Reply:
I think though that There is no need to do this any more as code you write inside new macro is fundamentally different. If you want to declare content with default values you write just name of the content to the correct section. The Constructor is inferred from parent type section. If macro documentation is not clear I can improve it.

Reply:
>  If you want to declare content with default values you write just name of the content to the correct section.

I don't understand what this means, and the examples don't show any default values being used (aside from `initializer`, which is not a replacement).